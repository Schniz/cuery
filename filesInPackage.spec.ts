import { spawnSync } from "child_process";

test("Files that will be published to the repo", () => {
  const result = spawnSync("npm", ["pack", "--dry-run"], { encoding: "utf8" });
  const lines = result.stderr.split("\n");
  const begin = lines.findIndex(line => line.includes("Tarball Contents"));
  const end = lines.findIndex(line => line.includes("Tarball Details"));
  const files = lines.slice(begin + 1, end).map(line => {
    return line.split(/\s+/)[3];
  });
  expect(files).toMatchSnapshot();
});
