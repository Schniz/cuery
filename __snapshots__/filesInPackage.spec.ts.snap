// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Files that will be published to the repo 1`] = `
Array [
  "package.json",
  "core.d.ts",
  "index.d.ts",
  "LICENSE",
  "mysql.d.ts",
  "README.md",
  "tsconfig.json",
]
`;
